packages/pigeon/platform_tests/android_unit_tests/android/app/src/test/java/com/example/android_unit_tests/AllDatatypesTest.java

```diff
@@ -0,0 +1,113 @@
+        .when(binaryMessenger)
+        .send(anyString(), any(), any());
+    FlutterEverything api = new FlutterEverything(binaryMessenger);
+    boolean[] didCall = {false};
+    api.echo(
+        everything,
+        (result) -> {
+          didCall[0] = true;
+          assertNull(everything.getABool());
+          assertNull(everything.getAnInt());
+          assertNull(everything.getADouble());
+          assertNull(everything.getAString());
+          assertNull(everything.getAByteArray());
+          assertNull(everything.getA4ByteArray());
+          assertNull(everything.getA8ByteArray());
+          assertNull(everything.getAFloatArray());
+          assertNull(everything.getAList());
+          assertNull(everything.getAMap());
+        });
+    assertTrue(didCall[0]);
+  }
+
+  private static HashMap makeMap(String key, Integer value) {
+    HashMap result = new HashMap();
+    result.put(key, value);
+    return result;
+  }
+
+  private static long[] floatToIntArray(double[] input) {
+    long[] result = new long[input.length];
+    for (int i = 0; i < input.length; ++i) {
+      result[i] = Math.round(input[i]);
+    }
+    return result;
+  }
+
+  @Test
+  public void hasValues() {
+    Everything everything = new Everything();
+    everything.setABool(false);
+    everything.setAnInt(1234L);
+    everything.setADouble(2.0);
+    everything.setAString("hello");
+    everything.setAByteArray(new byte[] {1, 2, 3, 4});
+    everything.setA4ByteArray(new int[] {1, 2, 3, 4});
+    everything.setA8ByteArray(new long[] {1, 2, 3, 4});
+    everything.setAList(Arrays.asList(new int[] {1, 2, 3}));
+    everything.setAMap(makeMap("hello", 1234));
+    BinaryMessenger binaryMessenger = mock(BinaryMessenger.class);
+    doAnswer(
+            invocation -> {
+              ByteBuffer message = invocation.getArgument(1);
+              BinaryMessenger.BinaryReply reply = invocation.getArgument(2);
+              message.position(0);
+              reply.reply(message);
+              return null;
+            })
+        .when(binaryMessenger)
+        .send(anyString(), any(), any());
+    FlutterEverything api = new FlutterEverything(binaryMessenger);
+    boolean[] didCall = {false};
+    api.echo(
+        everything,
+        (result) -> {
+          didCall[0] = true;
+          assertEquals(everything.getABool(), result.getABool());
+          assertEquals(everything.getAnInt(), result.getAnInt());
+          assertEquals(everything.getADouble(), result.getADouble());
+          assertEquals(everything.getAString(), result.getAString());
+          assertArrayEquals(everything.getAByteArray(), result.getAByteArray());
+          assertArrayEquals(everything.getA4ByteArray(), result.getA4ByteArray());
+          assertArrayEquals(everything.getA8ByteArray(), result.getA8ByteArray());
+          assertArrayEquals(
+              floatToIntArray(everything.getAFloatArray()),
+              floatToIntArray(result.getAFloatArray()));
```

Comment:
The missing float array was an oversight, added it.

The goal wasn't to hand roll a delta test, but since I knew we wanted to test equality exactly I wanted integer float values to avoid any rounding errors.  The only reason I rounded is because there is no way to straight cast from float to integers in java.

I can put `1/2^n` in the fractional part and I should be able to do equality testing too and give the coverage that we aren't somehow truncating the fractional part.

Reply:
done